src/diagram-api/diagramAPI.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { detectType } from './detectType';
+import { getDiagram, registerDiagram } from './diagramAPI';
+
+describe('DiagramAPI', () => {
+  it('should return default diagrams', () => {
+    expect(getDiagram('sequence')).not.toBeNull();
+  });
+
+  it('should throw error if diagram is not defined', () => {
+    expect(() => getDiagram('loki')).toThrow();
+  });
+
+  it('should handle diagram registrations', () => {
+    expect(() => getDiagram('loki')).toThrow();
+    // TODO Q: Shouldn't this be throwing an error?
+    expect(detectType('loki diagram')).toBe('flowchart');
```

Comment:
@knsv, this currently doesn't throw because the last case of detectType is 'flowchart'. 
I feel we should really be throwing an error instead, `Unable to determine diagram type`.